Rogue-Ape-Studios-XR/Secrets-of-Ignacios
# ✨ Spell sequence system

## Content

A sequence can now be made with hand gestures, it will recognise the spell it is supposed to be and print it in the console. The current spells and gestures are test gestures, these can be deleted later, no changes to the scripts are necessary.

## Changes
### Added
`Assets/Scripts/Spells/ElementType.cs` : Was created. Enum for spell element types.
`Assets/Scripts/Spells/Gesture.cs` : Was created. Scriptable object for gestures.
`Assets/Scripts/Spells/HandShape.cs` : Was created. Enum for all possible hand shapes.
`Assets/Scripts/Spells/ScriptableObjects/Gestures/` : Was created. Contains scriptable objects for gestures.
`Assets/Scripts/Spells/ScriptableObjects/Spells/Test.asset` : Was created. Test spell scriptable object.
`Assets/Scripts/Spells/SequenceManager.cs` : Was created. Manages when the sequence is started, creates the sequence and validates hand gestures.
`Assets/Scripts/Spells/Spell.cs` : Was created. Spell scriptable object script.
`Assets/Scripts/Spells/SpellManager.cs` : Was created. Manager for the spells the player is using, is for activating not casting at the moment.
`File` : Was created. You can add the purpose here.

### Updated
`Assets/Prefabs/PlayerRig/MR Interaction Setup.prefab` : Was updated / renamed. Added sequences to the player rig.

## Testing

The feature / Bugfix can be testing by following these steps:

1. Make sure the prefab is updated to the newest version.
2. Use the only valid sequence (thumbs up, fist, palm up), you can see the gesture is recognised in the console.
3. When the sequence is done it should print "HEEEEEEEEEEEEEEEEEEY [spell.name]" in the console.

Pls check for bugs I can't find any for now.

![Sequence Recognition Small](https://github.com/user-attachments/assets/49a2fbf3-9541-4a46-95bf-73a270f90b0e)

Closes #15
